operator/v1/types_klusterlet.go

```diff
@@ -162,6 +163,51 @@ type RegistrationConfiguration struct {
 	// +optional
 	// +kubebuilder:default:=100
 	KubeAPIBurst int32 `json:"kubeAPIBurst,omitempty"`
+
+	// PriorityBootstrapKubeconfigs defines the list of bootstrap kubeconfigs in the order of priority.
+	// They are used for an agent to connect to multiple hubs in cases like auto-failover, backup&restore, rollout upgrade etc.
+	//
+	// When the agent loses the connection to the current hub over HubConnectionTimeoutSeconds, or the managedcluster CR
+	// is set `hubAcceptsClient=false` on the hub, we mark the related bootstrap secret as failed.
+	// A failed bootstrapkubeconfig won't be used for the duration specified by SkipFailedBootstrapKubeconfigSeconds.
+	//
+	// Then the agent will try to choose another bootstrapkubeconfig from the list by order, replace the current bootstrapkubeconfig and trigger the rebootstrap.
+	// +optional
+	PriorityBootstrapKubeconfigs *PriorityBootstrapKubeconfigs `json:"priorityBootstrapKubeconfigs,omitempty"`
+
+	// HubConnectionTimeoutSeconds is used to set the timeout of connecting to the hub cluster.
+	// When agent loses the connection to the hub over the timeout seconds, the agent do a rebootstrap.
+	// By default is 10 mins.
+	// +optional
+	// +kubebuilder:default:=600
+	// +kubebuilder:validation:Minimum=0
+	HubConnectionTimeoutSeconds int32 `json:"hubConnectionTimeoutSeconds,omitempty"`
+
+	// SkipFailedBootstrapKubeconfigSeconds. The controller will skip the bootstrapkubeconfigs that fails recently.
+	// By default is 3 mins.
+	// +optional
+	// +kubebuilder:default:=180
+	// +kubebuilder:validation:Minimum=0
+	SkipFailedBootstrapKubeconfigSeconds int32 `json:"skipFailedBootstrapKubeconfigSeconds,omitempty"`
+}
+
+type TypePriorityBootstrapKubeconfigs string
+
+const (
+	LocalSecrets TypePriorityBootstrapKubeconfigs = "LocalSecrets"
+)
+
+type PriorityBootstrapKubeconfigs struct {
+	// Type specifies the type of priority bootstrap kubeconfigs.
+	// By default, it is set to LocalSecrets.
+	// +required
+	// +kubebuilder:default:=LocalSecrets
+	Type TypePriorityBootstrapKubeconfigs `json:"type,omitempty"`
+
+	// LocalSecrets is a list of secrets that contains the kubeconfigs for priority bootstrap.
+	// The secrets must be in the same namespace where the agent controller runs.
+	// +optional
+	LocalSecrets []corev1.LocalObjectReference `json:"localSecrets,omitempty"`
```

Comment:
I am not sure whether we want user to specify the secret name or we just have a fixed secret name format with certain label.